docs/tools/platformio.mdx

```diff
@@ -0,0 +1,124 @@
+
+## PlatformIO's project structure
+
+Every PlatformIO's project has a <a href="https://docs.platformio.org/en/latest/projectconf/section_platformio.html#directory-options" target="_blank">basic folder project structure<IconExternalLink width="10"/></a> and must have at the root of the project the file <a href="https://docs.platformio.org/en/latest/projectconf/section_platformio.html#generic-options" target="_blank">_platformio.ini_ <IconExternalLink width="10" /></a>.
+
+Checkout the Luos tutorial on <a href="https://www.luos.io/tutorials/pio" target="_blank">how to create a Luos project with PlatformIO <IconExternalLink width="10" /></a>.
+
+## Luos configuration for _platformio.ini_
+
+Some build flags definition may be used to configure Luos engine when it is added in your project. Those definitions will take care of the Luos engine library by using a Luos script called when building a project with Luos engine. Those definitions must be set in _platformio.ini_.
+For more information, see the script _source_filter_script.py_ in <a href="https://github.com/Luos-io/luos_engine" target="_blank" rel="external nofollow">Luos engine <IconExternalLink width="10" /></a>.
+
+### Definition for Luos engine Library
+
+| Definition  |                                            Description |                                                                           Value                                                                            |
+| :---------: | -----------------------------------------------------: | :--------------------------------------------------------------------------------------------------------------------------------------------------------: |
+|   LUOSHAL   | Defines the HAL family to use with Luos engine library | <a href="https://github.com/Luos-io/luos_engine/tree/main/engine/HAL" target="_blank" rel="external nofollow"> Luos HAL <IconExternalLink width="10" /></a> |
+| NOTELEMETRY |                                     Disables telemetry |                                                                       On by default                                                                        |
+
+Every Luos project needs to have a _node_config.h_ file allowing to configure Luos engine's hardware dependencies and configure the library. See the [basics concept page](/docs/luos-technology/basics/orga) for more information.
+
+Here is a _platformio.ini_ project example:
+
+```AsciiDoc
+[env:nucleo_f072rb]
+platform = ststm32
+board = nucleo_f072rb
+framework = stm32cube
+
+build_unflags = -Os
+build_flags =
+    -include node_config.h
+    -DLUOSHAL=STM32F0
+    -O1
+lib_deps =
+    luos_engine
+```
+
+This project in the examples is Luos engine dependent. PlatformIO will download the last version of the library Luos engine from its registry. The file _node_config.h_ allowing users to configure Luos parameters and the hardware definition to be added to the project.
+
```

Comment:
```suggestion
Because this example is Luos engine dependent, PlatformIO will download the last version of the Luos engine library from its registry.
Here is the result of a good configuration of Luos engine during compilation:

```

Reply:
done